SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMaps/GeneratedTestMap.cpp

```diff
@@ -32,18 +33,32 @@ UGeneratedTestMap::UGeneratedTestMap(EMapCategory InMapCategory, FString InMapNa
 	bIsValidForGeneration = true;
 }
 
+void UGeneratedTestMap::Init(EMapCategory InMapCategory, FString InMapName)
+{
+	MapCategory = InMapCategory;
+	MapName = InMapName;
+	bIsValidForGeneration = true;
+}
+
 AActor* UGeneratedTestMap::AddActorToLevel(ULevel* Level, UClass* Class, const FTransform& Transform)
 {
 	return GEditor->AddActor(Level, Class, Transform);
 }
 
-bool UGeneratedTestMap::GenerateMap()
+void UGeneratedTestMap::SetMapCategory(const EMapCategory InMapCategory)
+{
+	checkf(bIsGeneratingMap,
+		   TEXT("SetMapCategory should only be called between GenerateMap and SaveMap - most likely within CreateCustomContentForMap."));
+	MapCategory = InMapCategory;
+}
+
+void UGeneratedTestMap::GenerateMap()
 {
 	checkf(bIsValidForGeneration, TEXT("This test map object is not valid for map generation, please use the UGeneratedTestMap constructor "
 									   "with arguments when deriving from the base UGeneratedTestMap."));
+	bIsGeneratingMap = true;
```

Comment:
Curious why bIsGeneratingMap is set to true here, but not set to false at the bottom of this function?

Reply:
Agree with Martin that this is error prone, creates non obvious ordered dependencies, would prefer a different way of architecting this. Maybe ASpatialFunctionalTest::GenerateMap() could toggle a bGeneratingMap boolean on UGeneratedTestMap directly.

```
void ASpatialFunctionalTest::GenerateMap() 
{
	GeneratedTestMap->CurrentlyGeneratingFromTest(true); 
	GeneratedTestMap->GenerateMap(); 
	CreateCustomContentForMap(); 
	GeneratedTestMap->AddActorToLevel(GeneratedTestMap->GetWorld()>GetCurrentLevel(), this->GetClass(), FTransform(TestPositionInWorld)); 
	GeneratedTestMap->CurrentlyGeneratingFromTest(false); 
}
```